src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let lastResult = initialValue;
+    let i = 0;
```

Comment:
Use separate variables for logic and loops.

```suggestion
    let startIndex = 0;
```